src/chromium/crPage.ts

```diff
@@ -649,8 +649,18 @@ class FrameSession {
     this._page._onFileChooserOpened(handle);
   }
 
-  _onDownloadWillBegin(payload: Protocol.Page.downloadWillBeginPayload) {
-    this._crPage._browserContext._browser._downloadCreated(this._page, payload.guid, payload.url);
+  async _onDownloadWillBegin(payload: Protocol.Page.downloadWillBeginPayload) {
+    let originPage = this._crPage._initializedPage;
+    // If it's a new window download, report it on the opener page.
+    if (!originPage) {
+      originPage = await this._crPage.opener();
```

Comment:
this now races with the download finished, i.e. crashes

Reply:
Done.